### PR TITLE
Sync implementation

### PR DESCRIPTION
**Pull request by mtnt**

### CODE REVIEW

**Comment by mtnt** [comment]
@LinusU could you review it? or just tell me, what else I should todo - it is not very convenient to use forked repo

**Comment by KyleAMathews** [comment]
Please review and add this — right now the docs are wrong which was confusing for a while :-)

**Comment by roryrjb** [comment]
@Mtnt if you could also update the tests to include the `sync` implementation.

File: index.js [hunk 1]
```diff
@@ -43,4 +43,45 @@ function md5Dir (dirname, cb) {
...
 }
 
+function md5DirSync (dirpath, ignorePaths) {
+  if (!fs.existsSync(dirpath)) {
+    throw new Error(dirpath + ' does not exist.')
+  } else if (!fs.statSync(dirpath).isDirectory()) {
+    throw new Error(dirpath + ' is not a directory.')
+  }
+
+  if (ignorePaths && Object.prototype.toString.call(ignorePaths).slice(8, -1) !== 'Array') {
```
Comment (LinusU) [inline index.js]: You can use `Array.isArray()`?

**Comment by mtnt** [comment]
oh, sorry =) I will work on it at weekend.

**Comment by mtnt** [comment]
Huh! The pr was created a year ago... sorry for that, I was busy a little)
So, I fix several errors and add a test.

**Comment by mtnt** [comment]
@LinusU could you pay some attention?)

**Comment by LinusU** [comment]
I think that we should make `ignorePaths` part of an options object instead of its own argument, otherwise it will be very hard to add other options in the future.

Could we also add `ignorePaths` to the async version? I think that we should aim for feature parity between the two

**Comment by mtnt** [comment]
@LinusU check it out

**Comment by LinusU** [comment]
Looks good 👍 

Could you update the readme to reflect that `options` is an object

**Comment by mtnt** [comment]
As you wish =)

**Comment by mtnt** [comment]
@LinusU

**Comment by mtnt** [comment]
@LinusU could you finally merge it? I no longer can keep the fork.

**Comment by LinusU** [comment]
I have submitted a synchronous implementation in #3 

The problem with this PR is that it tried to piggyback in a `ignorePaths` option that would have been very nice to discuss prior to this.

I'm still open to adding a ignore option, but I would like the syntax of it to be discussed first. Ideally, I think that it should follow how `.gitignore`-files works, or similar. I believe that there are some good implementations of this that we could use on npm..

**Comment by LinusU** [comment]
Sync implementation is published as `v0.1.1`

**Comment by LinusU** [comment]
This seems very nice: https://www.npmjs.com/package/ignore

**Comment by LinusU** [comment]
See #4 for ignore support

**Pull request closed without merging**